### ISSUE TITLE
Discord?

### ISSUE THREAD

**Issue by AlexGipp** [aherd2985/UtilityBelt#86]
Wouldnt it be nice to make this program a discord bot?

Right now this program is not really meant for "regular" users so wouldnt it be nice to turn this into a discord bot?

**Comment by 1RedOne** [comment]
See [this functionality](https://github.com/aherd2985/UtilityBelt/blob/master/Utilities/DiscordSender.cs) under `Utilities\DiscordSender.cs`

**Comment by aherd2985** [comment]
If you want to go a different route than webhooks, you may be able to implement something like this:

https://docs.stillu.cc/guides/getting_started/first-bot.html

If we had something like that, we could tie it to an AI and create a chat or QnA bot:

https://www.qnamaker.ai/

**Comment by AlexGipp** [comment]
Yeah that was exactly what i was thinking about, do you think it would be a good idea to rewrite the program and make it a propper discord bot?

**Comment by aherd2985** [comment]
Could the new bot be created as a utility and built out on its own?

Trying to keep this as an open playground that anyone can add to.

**Comment by AlexGipp** [comment]
We could make a "main project" which contains all the discord related stuff and then a "module" project which contains all the commands and functionallity. That would still allowe people to easily add new module / functions.

What do you think about that?

**Comment by ArturLavrov** [comment]
@AlexGipp I think that we could combine both approaches.
From my perspective we could slightly reorganise solution with a following structure:

- UtilityBelt.Core

This layer will contains all main thing. For instance logging, how we load utilities and so on.
Also core utils could lives here.

- UtilityBelt.Console

This will be good old console app with existing command line interface.

- UtilityBelt.DiscordBot 

Will be your implementation of app functionality for Discord

What do you think about?

**Comment by AlexGipp** [comment]
the thing is that all the modules dont return a string for example, they all Console Writeline so we would definetly rewrite those a bit @ArturLavrov

**Comment by AlexGipp** [comment]
@aherd2985 i think the smartest thing would be to just create a second branch for the rework

**Comment by aherd2985** [comment]
Great idea @ArturLavrov 

That approach would be more scalable and keep the spirit of it being open-ended. 

I'd like this repo to have a solid foundation that encourages modular building. Hopefully we can keep it easy for contributors but also versatile enough to support growth in any direction.
